bouncer/src/repo/core/model/bson/repo_bson.h

```diff
@@ -54,7 +54,7 @@ namespace repo {
 			//TODO: Eventually we should inherit from a generic BSON object.
 			//work seems to have been started in here:https://github.com/jbenet/bson-cpp
 			//alternatively we can use a c++ wrapper on https://github.com/mongodb/libbson
-			class REPO_API_EXPORT RepoBSON : private mongo::BSONObj
+			class REPO_API_EXPORT RepoBSON : public mongo::BSONObj
```

Comment:
@har00n-haider  this is private on purpose - so we don't use the functions from the parents.

Where is it complaining?

Reply:
The full error is the commit message here:

https://github.com/3drepo/3drepobouncer/pull/546/commits/bf7ce11b63bf64df4bc0c5dfa648956188fa8e52

Reply:
I feel like it might be because we are printing the BSON somewhere using gtests print functions, I'll try narrowing it down

Reply:
sounds like we're iterating on it somewhere - can we fix it?

(If you ran out of time don't worry)

Reply:
I could not get to a full solution, but this is what I have found out:

- There is a default template print function that is getting called by our test code 
- Using gtest macros like ASSERT_EQ require a method to print different types to an outputstream
- we don't have a 'foo::PrintTo(const T&, ostream*)' or 'operator<<(ostream&, const T&)' defined for the repoBson object (as required gtest functions [here](https://github.com/google/googletest/blob/100f6fbf5f81a82d163c1e29735e8a2936eacd4f/googletest/include/gtest/gtest-printers.h#L43))
- gtest suite uses its own template function, which causes the compilation error

I've pushed a commit to revert the change to the inhertance accessor, hopefully this is enough to debug the issue!